app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -0,0 +1,47 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    # Generic JWT authenticator that receive JWT vendor configuration and uses to validate that the authentication
+    # request is valid, and return conjur authn token accordingly
+    JwtAuthenticate ||= CommandClass.new(
+      dependencies: {
+        token_factory: TokenFactory.new
+      },
+      inputs: %i[jwt_configuration account username]
+    ) do
+      def call
+        validate_and_decode
+        get_identity
+        validate_restrictions
+        get_token
+      end
+
```

Comment:
the `call` method should be the only public method

Reply:
also - please order the method implementations in the order they are called above

Reply:
Added private and reordered